[WIP] BZ-2082671: Removing networkConfig spec from BareMetalHost definition

abhatt-rh opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OSDOCS-<jira-issue-id>]: <short-description-of-the-pr> --->

Version(s): OpenShift 4.10+
<!--- Specify the version or versions of OpenShift your PR applies to.
If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch. --->

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2082671
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: http://file.emea.redhat.com/avbhatt/bz-2082671/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster.html#preparing-the-bare-metal-node_ipi-install-expanding
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build.
  - The preview will be generated after you open the PR.
  - You will need to edit the comment to add the link after the preview builds.
  - Review the preview build to make sure your changes are rendering properly. --->

Additional information: Updated the docs to remove the `networkConfig` spec from `BareMetalHost` definition since it is not included in the API reference documentation for [BareMetalHost [metal3.io/v1alpha1]](https://docs.openshift.com/container-platform/4.10/rest_api/provisioning_apis/baremetalhost-metal3-io-v1alpha1.html) 

Also updated the YAML to reflect the correct `BareMetalHost` spec for configuring network (preprovisioningNetworkDataName)


<!--- Optional: Include additional context or expand the description here.--->



<!--- Next steps after opening your PR:

* Ask for peer review from the OpenShift docs team:
  - For Red Hat associates: Ping @peer-review-squad requesting a review in the #forum-docs-review channel (CoreOS Slack workspace) and provide the following information:
    * A link to the PR.
    * The size of the PR that the GitHub bot assigns (ex: XS, S, M, L, XL).
    * If there is urgency or a deadline for the review.
  - For community authors: Request a review by tagging @openshift/team-documentation in a GitHub comment.

  Slack is the quickest and preferred way to request a review.

* IMPORTANT:
  - All documentation changes must be verified by a QE team associate before merging.
  - Squash to one commit before submitting your PR for peer review.

* For more information about verifying your content, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#verification-of-your-content

* For more information about contributing to OpenShift documentation, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/contributing.adoc

Additional resources

The OpenShift docs repo adheres to the following style guides:

- OpenShift documentation guidelines (OSDOCS)
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc
- Red Hat Supplementary Style Guide (SSG)
  https://redhat-documentation.github.io/supplementary-style-guide/
- Modular Documentation Reference Guide (Mod Docs)
  https://redhat-documentation.github.io/modular-docs/
- IBM Style Guide (ISG)
  https://www.ibm.com/docs/en/ibm-style

You can log in to the ISG by using your @redhat.com id and single sign-on (SSO) credentials. --->

abhatt-rh commented:
Hi @nstamatelopoulos thank you for providing a thorough review during our conversation. I have updated the docs per your suggestions.
@nstamatelopoulos and @awolffredhat,
Request you to kindly review the updated docs and provide your inputs and approval.

awolffredhat commented:
1. It needs to be clarified that the nmstate configuration in the secret is an example. Other network configurations can be set, too.
2. Since other configuration items are generalized and put into brackets <>, I think the ones in the network secret should be generalized, too (e.g. ip address)

abhatt-rh commented:
Hi @ktothill,
Adding you as reviewer here to avoid any redundancies in content coverage.

awolffredhat commented on modules/ipi-install-preparing-the-bare-metal-node.adoc:
```diff
@@ -53,52 +53,62 @@ $ vim bmh.yaml
-    interfaces:
-    - name: <NIC1_name>
-      type: ethernet
+ kind: Secret
```
It is important to note that the network configuration is optional. The secret and the reference to it only need to be added if the user wants to add network settings to the newly created node.

abhatt-rh replied:
Revised the [callout entry](https://github.com/openshift/openshift-docs/pull/45918/files#diff-f90b46cde12693fcb811ca202b3759cea7652922c2b7f92da22716e5e8aaa180R105).

abhatt-rh commented:
> 1. It needs to be clarified that the nmstate configuration in the secret is an example. Other network configurations can be set, too.

I have revised the callout to clarify this. Kindly review.
> 2. Since other configuration items are generalized and put into brackets <>, I think the ones in the network secret should be generalized, too (e.g. ip address)

Based on this [OpenShift documentation guideline for IP addresses](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#ip-addresses), we can use the mentioned IP addresses without the brackets.

awolffredhat commented:
Thanks, Avani. Should the preprovisioningNetworkDataName line also say that it's optional? The secret should obviously only be referenced if it's created.

abhatt-rh commented:
> Thanks, Avani. Should the preprovisioningNetworkDataName line also say that it's optional? The secret should obviously only be referenced if it's created.

Thanks, @awolffredhat ! You're right. I have updated the step and marked it as optional. Kindly re-review and provide an approval/lgtm.

nstamatelopoulos commented:
Great work Avani. I think it is all included now in detail. Looks good for a merge to me too.

abhatt-rh commented:
Hi @johnwilkins. This PR updates a procedure in the bare metal IPI docs. Requesting your review too. Thanks!

awolffredhat commented:
I just ran the procedure on a real cluster and had to make some changes for the bmh to reach 'available' state. I used the following file. 
I believe the differences are:
1. Formatting was needed.
2. three fields needed to be added to bmc section:
```
    disableCertificateVerification: True
    username: admin
    password: password
```


```
apiVersion: v1
kind: Secret
metadata:
  name: openshift-worker-2-network-config-secret   
type: Opaque
stringData:
 nmstate: |
   routes:
     config:
     - destination: 0.0.0.0/0
       next-hop-address: 192.168.123.1
       next-hop-interface: enp0s4
   dns-resolver:
     config:
       server:
       - 192.168.123.1
   interfaces:
   - name: enp0s4
     state: up
     ipv4:
       address:
       - ip: 192.168.123.16
         prefix-length: 24
       enabled: true
       dhcp: false
---
apiVersion: v1
kind: Secret
metadata:
  name: openshift-worker-2-bmc-secret 
  namespace: openshift-machine-api
type: Opaque
data:
  username: YWRtaW4=
  password: cGFzc3dvcmQ=
---
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: openshift-worker-2 
  namespace: openshift-machine-api
spec:
  online: true
  bootMACAddress: 52:54:00:8e:ed:09
  bmc:
    address: redfish://192.168.123.1:8000/redfish/v1/Systems/7d88ebf8-beac-4c90-b77e-5cd620a2b023
    credentialsName: openshift-worker-2-bmc-secret
    disableCertificateVerification: True
    username: admin
    password: password
  preprovisioningNetworkDataName: openshift-worker-2-network-config-secret

```

abhatt-rh commented:
Hi @awolffredhat, per https://github.com/openshift/openshift-docs/pull/45918#issuecomment-1141344629, I have added the three fields and updated the formatting. Pls review

awolffredhat commented:
Thanks, Avani. 
LGTM

michaelryanpeter commented on modules/ipi-install-preparing-the-bare-metal-node.adoc:
```diff
@@ -53,52 +53,68 @@ $ vim bmh.yaml
+    password: <bmc-password> <9>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <10> <2>
+----
+<1> (Optional) To configure network for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node.
```
```suggestion
<1> Optional: To configure network for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node.
```
re: [ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps)

michaelryanpeter commented on modules/ipi-install-preparing-the-bare-metal-node.adoc:
```diff
@@ -53,52 +53,68 @@ $ vim bmh.yaml
+<7> To skip certificate validation, set `disableCertificateVerification` to `true`.
+<8> Replace `<bmc-username>` with the `bmc` string of the user name.
+<9> Replace `<bmc-password>` with the `bmc` string of the password.
+<10> (Optional) Provide the network configuration secret name in the `preprovisioningNetworkDataName` of the `BareMetalHost` CR if you have configured network for the newly created node.
```
```suggestion
<10> Optional: Provide the network configuration secret name in the `preprovisioningNetworkDataName` of the `BareMetalHost` CR if you have configured network for the newly created node.
```
See above.

michaelryanpeter commented on installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster.adoc:
```diff
@@ -19,6 +19,7 @@ include::modules/ipi-install-preparing-the-bare-metal-node.adoc[leveloffset=+1]
 .Additional resources
 
 * xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[(Optional) Configuring host network interfaces in the install-config.yaml file]
```
```suggestion
* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[Optional: Configuring host network interfaces in the install-config.yaml file]
```

abhatt-rh replied:
I'd like skip this one for now since it requires me to change another module's heading, which lies under a chapter with such similar headings. Furthermore, this module has another instance of the usage of "(Optional)" [here](https://github.com/openshift/openshift-docs/pull/45918/files#diff-f90b46cde12693fcb811ca202b3759cea7652922c2b7f92da22716e5e8aaa180R14). It makes sense to change all of them or none of them.

johnwilkins replied:
@abhatt-rh I think you can skip that for now. There are quite a few other titles with (Optional) rather than Optional: I can make those changes in a separate PR so there is consistency across the IPI doc set.

michaelryanpeter commented on modules/ipi-install-preparing-the-bare-metal-node.adoc:
```diff
@@ -53,52 +53,68 @@ $ vim bmh.yaml
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <10> <2>
+----
+<1> (Optional) To configure network for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node.
+<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, `credentialsName` field, and the `preprovisioningNetworkDataName` field.
```
```suggestion
<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, the `credentialsName` field, and the `preprovisioningNetworkDataName` field.
```

michaelryanpeter reviewed (commented):
I just left a couple of small comments. Let me if you have questions.

abhatt-rh commented on modules/ipi-install-preparing-the-bare-metal-node.adoc:
```diff
@@ -53,52 +53,68 @@ $ vim bmh.yaml
+<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, `credentialsName` field, and the `preprovisioningNetworkDataName` field.
+<3> Replace `<base64-of-uid>` with the `base64` string of the user name.
+<4> Replace `<base64-of-pwd>` with the `base64` string of the password.
+<5> Replace `<NIC1-mac-address>` with the MAC address of the bare metal node's first NIC. See the BMC addressing section for additional BMC configuration options. 
```
@michaelryanpeter Good point. Instead of updating in the call out, I defined the acronym at it's first usage [here](https://github.com/openshift/openshift-docs/pull/45918/files#diff-f90b46cde12693fcb811ca202b3759cea7652922c2b7f92da22716e5e8aaa180R33).

michaelryanpeter commented:
/lgtm

michaelryanpeter commented:
/cherrypick enterprise-4.10
